storage/flux/table.gen.go

```diff
@@ -801,6 +781,113 @@ func (t *floatGroupTable) advance() bool {
 	return true
 }
 
+type floatAggregateMethod func([]int64, []float64) (int64, float64)
+
+// determineFloatAggregateMethod returns the method for aggregating
+// returned points within the same group. The incoming points are the
+// ones returned for each series and the method returned here will
+// aggregate the aggregates.
+func determineFloatAggregateMethod(agg datatypes.Aggregate_AggregateType) (floatAggregateMethod, error) {
+	switch agg {
+	case datatypes.AggregateTypeFirst:
+		return groupByFirstFloat, nil
+	case datatypes.AggregateTypeLast:
+		return groupByLastFloat, nil
+	case datatypes.AggregateTypeCount:
+
+		return nil, &influxdb.Error{
+			Code: influxdb.EInvalid,
+			Msg:  "unsupported for aggregate count: Float",
+		}
+
+	case datatypes.AggregateTypeSum:
+
+		return groupBySumFloat, nil
+
+	case datatypes.AggregateTypeMin:
+
+		return groupByMinFloat, nil
+
+	case datatypes.AggregateTypeMax:
+
+		return groupByMaxFloat, nil
+
+	default:
+		return nil, &influxdb.Error{
+			Code: influxdb.EInvalid,
+			Msg:  fmt.Sprintf("unknown/unimplemented aggregate type: %v", agg),
+		}
+	}
+}
+
+func groupByMinFloat(timestamps []int64, values []float64) (int64, float64) {
+	value := values[0]
+	timestamp := timestamps[0]
+
+	for i := 1; i < len(values); i++ {
+		if value > values[i] {
+			value = values[i]
+			timestamp = timestamps[i]
+		}
+	}
+
+	return timestamp, value
+}
+
+func groupByMaxFloat(timestamps []int64, values []float64) (int64, float64) {
+	value := values[0]
+	timestamp := timestamps[0]
+
+	for i := 1; i < len(values); i++ {
+		if value < values[i] {
+			value = values[i]
+			timestamp = timestamps[i]
+		}
+	}
+
+	return timestamp, value
+}
+
+// For group count and sum, the timestamp here is always math.MaxInt64.
+// their final result does not contain _time, so this timestamp value can be anything
+// and it won't matter.
+
+func groupBySumFloat(timestamps []int64, values []float64) (int64, float64) {
```

Comment:
Typically this is something that would benefit from going as fast as possible, and it shouldn't be too complicated to improve performance for a nice tight loop like this. 

One thing that might be worth trying here is to unroll the loop. Typically I have found with right loops like this you can usually improve performance by 30-40% depending on the situation. 

I haven't tested this but I would try something like:

```go
        // we need to know how many values might not fit into blocks of four.
       // we can sum them up afterwards.
        remainder := len(values) % 4

        // we will store the intermediate results in this array.
	var sums [4]float64

        // sum up any extra values...
	for _, v := range a[len(a)-remainder:] {
		sums[0] += v
	}
        
        // iterate over four values at a time. This can improve overall perf
        // because of a bunch of reasons (bounds checking, pipelining, better cache, fewers instructions). 
	for i := 0; i < len(a)-remainder; i += 4 {
		sums[0] += a[i+3] //  as a habit I put the biggest index first incase it helps reduce bounds checking...
		sums[1] += a[i+2]
		sums[2] += a[i+1]
		sums[3] += a[i]
	}

        // just need to sum up the intermediate results
        return math.MaxInt64, sums[0] + sums[1] + sums[2] + sums[3]
```

Of course, I'm speculating and benchmarks are the source of truth 😄. But I would expect something like the above to be quite a bit faster for larger groups of values.

Reply:
This is a cool idea and I'd be interested in benchmarking this as part of a future PR.